`libgcc_s.so.1 must be installed for pthread_cancel to work` during SCT's post-install validation

### Description

When submitting our `Shimming-Toolbox` Pull Request to the CI, we get an error that seems to be related to the installation of `SCT`. These tests were passing a few days ago and we have not modified our PR since then, so it seems to come from `SCT`'s end.

This happens with [this PR](https://github.com/shimming-toolbox/shimming-toolbox/pull/295/checks?check_run_id=3499425571) and [this one](https://github.com/shimming-toolbox/shimming-toolbox/actions/runs/1191672436). 

The bug happens when testing under `Ubuntu-18.04 3.8`.

<details>
  <summary>See error output</summary>
  
```
Spinal Cord Toolbox (git-master-666237aeb31246ff163905c7f3cf6cbe8d696e44)

sct_check_dependencies 
--

SCT info:
- version: git-master-666237aeb31246ff163905c7f3cf6cbe8d696e44
- path: /home/runner/work/spinalcordtoolbox
OS: linux (Linux-5.4.0-1056-azure-x86_64-with-debian-buster-sid)
CPU cores: Available: 2, Used by ITK functions: 2
RAM: Total: 6954MB, Used: 495MB, Available: 6128MB
Check Python executable.............................[OK]
  Using bundled python 3.6.13 |Anaconda, Inc.| (default, Jun  4 2021, 14:25:59) 
[GCC 7.5.0] at /home/runner/work/spinalcordtoolbox/python/envs/venv_sct/bin/python
Check if data are installed.........................[OK]
Check if colored is installed.......................[OK] (1.4.2)
Check if dipy is installed..........................[OK] (1.4.1)
Check if futures is installed.......................[OK]
Check if h5py is installed..........................[OK] (2.10.0)
Check if Keras (2.1.5) is installed.................[OK] (2.1.5)
Check if ivadomed is installed......................[OK] (2.8.0)
Check if matplotlib is installed....................[OK] (3.3.4)
Check if nibabel is installed.......................[OK] (3.2.1)
Check if numpy is installed.........................[OK] (1.19.5)
Check if onnxruntime (1.4.0) is installed...........[OK] (1.4.0)
Check if pandas is installed........................[OK] (1.1.5)
Check if psutil is installed........................[OK] (5.8.0)
Check if pyqt5 (5.11.3) is installed................[OK] (5.11.3)
Check if pytest is installed........................[OK] (6.2.5)
Check if pytest-cov is installed....................[OK] (2.12.1)
Check if raven is installed.........................[OK]
Check if requests is installed......................[OK] (2.26.0)
Check if requirements-parser is installed...........[OK] (0.2.0)
Check if scipy is installed.........................[OK] (1.5.4)
Check if scikit-image is installed..................[OK] (0.17.2)
Check if scikit-learn is installed..................[OK] (0.24.2)
Check if tensorflow (1.5.0) is installed............[OK] (1.5.0)
Check if torch (1.5.0+cpu) is installed.............[OK] (1.5.0+cpu)
Check if torchvision (0.6.0+cpu) is installed.......[OK] (0.6.0+cpu)
Check if xlwt is installed..........................[OK] (1.3.0)
Check if tqdm is installed..........................[OK] (4.62.2)
Check if transforms3d is installed..................[OK] (0.3.1)
Check if urllib3 is installed.......................[OK] (1.26.6)
Check if pytest_console_scripts is installed........[OK]
Check if wquantiles is installed....................[OK] (0.4)
Check if spinalcordtoolbox is installed.............[OK]
Check ANTs compatibility with OS ...................[OK]
Check PropSeg compatibility with OS ................[OK]
Check if figure can be opened with matplotlib.......Generating new fontManager, this may take some time...
libgcc_s.so.1 must be installed for pthread_cancel to work
./install_sct: line 738:  3219 Aborted                 (core dumped) sct_check_dependencies

Installation validation Failed!


Installation failed!

Please copy the output of this Terminal (starting with the command install_sct) and upload it as a .txt attachment in a new topic on SCT's forum:

--> http://forum.spinalcordmri.org/c/sct

Error: Process completed with exit code 1.
```
</details>


### Steps to Reproduce

Follow [this link](https://github.com/shimming-toolbox/shimming-toolbox/actions/runs/1191672436) and hit the `Re-run jobs` button.

**Expected behavior:** 

All the tests should pass. 

**Actual behavior:**

The `Ubuntu-18.04 3.8` test fails and cancels the execution of all the other tests.

--- a/spinalcordtoolbox/scripts/sct_check_dependencies.py
+++ b/spinalcordtoolbox/scripts/sct_check_dependencies.py
@@ -261,6 +261,10 @@ def main(argv=None):
     else:
         print_fail()
 
+    # Import matplotlib.pyplot here (before PyQt can be imported) in order to mitigate a libgcc error
+    # See also: https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3511#issuecomment-912167649
+    import matplotlib.pyplot as plt
+
     for dep_pkg, dep_ver_spec in get_dependencies():
         if dep_ver_spec is None:
             print_line('Check if %s is installed' % (dep_pkg))
@@ -322,10 +326,9 @@ def main(argv=None):
     print_line('Check if figure can be opened with matplotlib')
     try:
         import matplotlib
-        import matplotlib.pyplot as plt
         # If matplotlib is using a GUI backend, the default 'show()` function will be overridden
         # See: https://github.com/matplotlib/matplotlib/issues/20281#issuecomment-846467732
-        fig = plt.figure()
+        fig = plt.figure()  # NB: `plt` was imported earlier in the script to avoid a libgcc error
         if getattr(fig.canvas.manager.show, "__func__", None) != matplotlib.backend_bases.FigureManagerBase.show:
             print_ok(f" (Using GUI backend: '{matplotlib.get_backend()}')")
         else:

